From c0165da97e4e3610c52bace632aa6e773ccd881b Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Mon, 19 Mar 2018 12:30:36 +0200
Subject: [PATCH] [hs] add CMakeLists.txt

---
 CMakeLists.txt | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..52aed04
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 2.6)
+
+set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+project(NXDNClients)
+
+file(GLOB GATEWAY_SOURCES "NXDNGateway/*.cpp")
+file(GLOB GATEWAY_HEADERS "NXDNGateway/*.h")
+
+file(GLOB PARROT_SOURCES "NXDNParrot/*.cpp")
+file(GLOB PARROT_HEADERS "NXDNParrot/*.h")
+
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Os -Wall -std=c++0x $ENV{CXXFLAGS}")
+set(DEPLIBS "")
+
+find_package(Threads REQUIRED QUIET)
+list(APPEND DEPLIBS ${CMAKE_THREAD_LIBS_INIT})
+
+add_executable(NXDNGateway ${GATEWAY_SOURCES} ${GATEWAY_HEADERS})
+target_link_libraries(NXDNGateway ${DEPLIBS})
+
+add_executable(NXDNParrot ${PARROT_SOURCES} ${PARROT_HEADERS})
+target_link_libraries(NXDNParrot ${DEPLIBS})
-- 
2.15.1

