From eef04dd9712af88cb9bf94ffcfe3c6a60749e6d9 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Mon, 19 Mar 2018 12:39:26 +0200
Subject: [PATCH] [hs] add CMakeLists.txt

---
 CMakeLists.txt | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..241b7fe
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 2.6)
+
+set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+project(P25Clients)
+
+file(GLOB GATEWAY_SOURCES "P25Gateway/*.cpp")
+file(GLOB GATEWAY_HEADERS "P25Gateway/*.h")
+
+file(GLOB PARROT_SOURCES "P25Parrot/*.cpp")
+file(GLOB PARROT_HEADERS "P25Parrot/*.h")
+
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Os -Wall -std=c++0x $ENV{CXXFLAGS}")
+set(DEPLIBS "")
+
+find_package(Threads REQUIRED QUIET)
+list(APPEND DEPLIBS ${CMAKE_THREAD_LIBS_INIT})
+
+add_executable(P25Gateway ${GATEWAY_SOURCES} ${GATEWAY_HEADERS})
+target_link_libraries(P25Gateway ${DEPLIBS})
+
+add_executable(P25Parrot ${PARROT_SOURCES} ${PARROT_HEADERS})
+target_link_libraries(P25Parrot ${DEPLIBS})
-- 
2.15.1

